src/stacked_borrows.rs

```diff
@@ -637,9 +613,9 @@ impl Stacks {
             Pointer::new(alloc_id, range.start),
             range.size.bytes()
         );
-        let global = &*state.borrow();
         self.for_each(range, move |offset, stack| {
-            stack.access(AccessKind::Read, tag, (alloc_id, range, offset), global)
+            let mut state = state.borrow_mut();
```

Comment:
Is it necessary to do `borrow_mut` inside the loop each time, rather than once outside?

Reply:
Only to comply with the `Fn` requirement of `for_each`. Moving it to `FnMut` fixes that, which seems spooky but we're already using `RefCell` so it's probably no more spooky actually.

Reply:
Yes, this is all sequential so `FnMut` makes a lot of sense actually.